src/access_nri_intake/data/utils.py

```diff
@@ -48,21 +48,41 @@ def available_versions(pretty: bool = True):
     # Work out where the catalogs are stored
     base_path = _get_catalog_rp()
 
+    # Grab the extant catalog and work out its min and max versions
+    try:
+        with open(get_catalog_fp()) as cat:
+            cat_yaml = yaml.safe_load(cat)
+            vers_min = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["min"]
+            vers_max = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["max"]
+            vers_def = cat_yaml["sources"]["access_nri"]["parameters"]["version"][
+                "default"
+            ]
+    except FileNotFoundError:
+        raise FileNotFoundError(f"Unable to find catalog at {get_catalog_fp()}")
+    except KeyError:
+        raise RuntimeError(f"Catalog at {get_catalog_fp()} not correctly formatted")
+
     # Grab all the catalog names
-    cats = [d for d in os.listdir(base_path) if re.search(CATALOG_NAME_FORMAT, d)]
+    cats = [
+        d.name
+        for d in base_path.iterdir()
+        if re.search(CATALOG_NAME_FORMAT, d.name)
+        and d.is_dir()
+        and ((d.name >= vers_min and d.name <= vers_max) or d.name == vers_def)
+    ]
     cats.sort(reverse=True)
 
     # Find all the symlinked versions
-    symlinks = [s for s in cats if os.path.islink(os.path.join(base_path, s))]
+    symlinks = [s for s in cats if (Path(base_path) / s).is_symlink()]
 
-    symlink_targets = {
-        s: os.path.basename(os.readlink(os.path.join(base_path, s))) for s in symlinks
-    }
+    symlink_targets = {s: (base_path / s).readlink().name for s in symlinks}
 
     if pretty:
-        for i, c in enumerate(cats):
+        for c in cats:
             if c in symlink_targets.keys():
                 c += f"(-->{symlink_targets[c]})"
+            if c == vers_def:
+                c += "*"
             print(c)
         return
 
```

Comment:
Hmmm I can't see any differences there, can you try this? (Note some variable name refactoring may be needed.)

```python
def available_versions(pretty: bool = True) -> list[str] | None:
    """
    Report the available versions of the `intake.cat.access_nri` catalog.

    Parameters
    ---------
    pretty : bool, optional
        Defines whether to return a pretty print-out of the available versions
        (True, default), or to provide a list of version numbers only (False).
    """
    # Work out where the catalogs are stored
    base_path = _get_catalog_root()

    # Grab the extant catalog and work out its min and max versions
    try:
        with open(get_catalog_fp()) as c:
            cat_yaml = yaml.safe_load(c)
            vers_min = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["min"]
            vers_max = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["max"]
            vers_def = cat_yaml["sources"]["access_nri"]["parameters"]["version"][
                "default"
            ]
    except FileNotFoundError:
        raise FileNotFoundError(f"Unable to find catalog at {get_catalog_fp()}")
    except KeyError:
        raise RuntimeError(f"Catalog at {get_catalog_fp()} not correctly formatted")

    # Grab all the catalog names
    cats = [
        dir_path.name
        for dir_path in base_path.iterdir()
        if re.search(CATALOG_NAME_FORMAT, dir_path.name)
        and dir_path.is_dir()
        and (
            (dir_path.name >= vers_min and dir_path.name <= vers_max)
            or dir_path.name == vers_def
        )
    ]
    cats.sort(reverse=True)

    # Find all the symlinked versions
    symlinks = [s for s in cats if (Path(base_path) / s).is_symlink()]

    symlink_targets = {s: (base_path / s).readlink().name for s in symlinks}

    if pretty:
        for c in cats:
            if c in symlink_targets.keys():
                c += f"(-->{symlink_targets[c]})"
            if c == vers_def:
                c += "*"
            print(c)
        return None

    return cats
```

Reply:
I'm pretty sure this is because of the repeated variable name:
```python
53   with open(get_catalog_fp()) as c:
54            cat_yaml = yaml.safe_load(c)
...
```
and
```python
84    if pretty:
85        for c in cats:
```

Mypy is inferring that these are the same variable - hence the error. If you change

```python
-        with open(get_catalog_fp()) as c:
-            cat_yaml = yaml.safe_load(c)

+          with open(get_catalog_fp()) as cat_file:
+            cat_yaml = yaml.safe_load(cat_file)
```
or similar it should fix it.

Reply:
Alternatively, I think you can add a type hint comment - something like 
```python
        for c in cats: #type: string
```

Reply:
Yeah, it was the repeat variable name - got in OK now.